aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -164,7 +164,10 @@ def should_run_final_scf(self):
         If the maximum number of meta convergence iterations has been exceeded and convergence has not been reached, the
         structure cannot be considered to be relaxed and the final scf should not be run.
         """
-        return self.ctx.is_converged and 'final_scf_inputs' in self.ctx
+        return (
+            RelaxType(self.inputs.relax_type) is not RelaxType.NONE and self.ctx.is_converged and
```

Comment:
Instead of converting `self.inputs.relax_type` into a `RelaxType` instance, we should just do it once in `setup` and assign it to the context.  While we are at it, we can move this code
```

        if 'relaxation_scheme' in self.inputs:
            if self.inputs.relaxation_scheme.value == 'relax':
                relax_type = RelaxType.ATOMS
            elif self.inputs.relaxation_scheme.value == 'vc-relax':
                relax_type = RelaxType.ATOMS_CELL
            else:
                raise ValueError('unsupported value for the `relaxation_scheme` input.')
        else:
            relax_type = RelaxType(self.inputs.relax_type)
```
from `run_relax` to setup and just assign `relax_type` to `self.ctx.relax_type`. Then everywhere else in the code you can just use `self.ctx.relax_type` and compare directly to `RelaxType`.